tools/codegen/gen_backend_stubs.py

```diff
@@ -30,6 +32,10 @@ def parse_backend_yaml(
 
     backend = yaml_values.pop('backend', None)
     assert backend is not None, 'You must provide a value for "backend"'
+    backend_key = DispatchKey.try_parse(backend)
```

Comment:
Can't you just rely on the parser failure propagating here?

Reply:
I totally can- I added this to make the error message clearer. That way if e.g., a backend without a corresponding autograd key tried to add an autograd entry, they'd get a more actionable error message than a parsing error.

Do you think the parsing error is good enough? We probably want good documentation on the new codegen anyway, that makes it clear how to opt into autograd kernels.

Reply:
Another way to make the error better is to "push an error context" before doing the parse. The preexisting example:

```
    with context(f'in native_functions.yaml line {f.loc}:\n  {f.func}'):
```